### PR TITLE
feat(k8s): add v1 and deprecate beta

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <pcyvoct@scaleway.com>

### CODE REVIEW

File: scaleway/helpers_k8s.go [hunk 1]
```diff
@@ -57,11 +58,43 @@ func k8sAPIWithRegionAndID(m interface{}, id string) (*k8s.API, scw.Region, stri
 	return k8sAPI, region, ID, err
 }
 
+func waitK8SClusterPoolRequired(k8sAPI *k8s.API, region scw.Region, clusterID string) error {
```
Comment (kindermoumoute) [inline scaleway/helpers_k8s.go:61]: Can use a single function to reach a desired list of state 
 `waitK8SClusterPool(k8sAPI *k8s.API, region scw.Region, clusterID string, desiredState ... k8s.PoolStatus)`

File: scaleway/resource_k8s_cluster_beta.go [hunk 2]
```diff
@@ -418,56 +430,66 @@ func resourceScalewayK8SClusterBetaCreate(d *schema.ResourceData, m interface{})
...
 
-		err = waitK8SPoolReady(k8sAPI, region, expandID(pool.ID))
+	} else {
+		err = waitK8SClusterPoolRequired(k8sAPI, region, res.ID)
 		if err != nil {
 			return err
 		}
 	}
 
+	d.SetId(newRegionalId(region, res.ID))
```
Comment (kindermoumoute) [inline scaleway/resource_k8s_cluster_beta.go:491]: Why do you do that here? This is done in the Read

File: scaleway/resource_k8s_cluster_beta_test.go [hunk 3]
```diff
@@ -6,352 +6,180 @@ import (
...
+					resource.TestCheckResourceAttr("scaleway_k8s_cluster.autoscaler", "tags.0", "terraform-test"),
+					resource.TestCheckResourceAttr("scaleway_k8s_cluster.autoscaler", "tags.1", "scaleway_k8s_cluster"),
+					resource.TestCheckResourceAttr("scaleway_k8s_cluster.autoscaler", "tags.2", "autoscaler-config"),
 				),
 			},
 		},
 	})
 }
 
-func TestAccScalewayK8SClusterBetaDefaultPool(t *testing.T) {
```
Comment (kindermoumoute) [inline scaleway/resource_k8s_cluster_beta_test.go:168]: Isn't it too soon to remove these tests? The attribute `default_pool` is still there even if it's deprecated

File: scaleway/resource_k8s_cluster_beta_test.go [hunk 4]
```diff
@@ -402,46 +230,34 @@ func testAccCheckScalewayK8SClusterBetaExists(n string) resource.TestCheckFunc {
...
-	tags = [ "terraform-test", "scaleway_k8s_cluster_beta", "minimal" ]
+	tags = [ "terraform-test", "scaleway_k8s_cluster", "minimal" ]
 }`, version)
 }
 
-func testAccCheckScalewayK8SClusterBetaConfigIngressDashboard(version string, ingress string, dashboard bool) string {
+func testAccCheckScalewayK8SClusterConfigIngressDashboard(version string, ingress string, dashboard bool) string {
 	return fmt.Sprintf(`
-resource "scaleway_k8s_cluster_beta" "ingressdashboard" {
+resource "scaleway_k8s_cluster" "ingressdashboard" {
```
Comment (kindermoumoute) [inline scaleway/resource_k8s_cluster_beta_test.go:245]: should keep `_beta`

File: scaleway/resource_k8s_cluster_beta_test.go [hunk 5]
```diff
@@ -402,46 +230,34 @@ func testAccCheckScalewayK8SClusterBetaExists(n string) resource.TestCheckFunc {
 	}
 }
 
-func testAccCheckScalewayK8SClusterBetaConfigMinimal(version string) string {
+func testAccCheckScalewayK8SClusterConfigMinimal(version string) string {
 	return fmt.Sprintf(`
-resource "scaleway_k8s_cluster_beta" "minimal" {
+resource "scaleway_k8s_cluster" "minimal" {
```
Comment (kindermoumoute) [inline scaleway/resource_k8s_cluster_beta_test.go:235]: should keep `_beta`

File: scaleway/resource_k8s_cluster_beta_test.go [hunk 6]
```diff
@@ -6,352 +6,180 @@ import (
...
 			},
 		},
 	})
 }
 
-func testAccCheckScalewayK8SClusterBetaDestroy(s *terraform.State) error {
+func testAccCheckScalewayK8SClusterDestroy(s *terraform.State) error {
 	for _, rs := range s.RootModule().Resources {
-		if rs.Type != "scaleway_k8s_cluster_beta" {
+		if rs.Type != "scaleway_k8s_cluster" {
```
Comment (kindermoumoute) [inline scaleway/resource_k8s_cluster_beta_test.go:182]: should keep `_beta`

File: scaleway/resource_k8s_cluster_beta_test.go [hunk 7]
```diff
@@ -402,46 +230,34 @@ func testAccCheckScalewayK8SClusterBetaExists(n string) resource.TestCheckFunc {
...
-	tags = [ "terraform-test", "scaleway_k8s_cluster_beta", "ingressdashboard" ]
+	tags = [ "terraform-test", "scaleway_k8s_cluster", "ingressdashboard" ]
 }`, version, ingress, dashboard)
 }
 
-func testAccCheckScalewayK8SClusterBetaConfigAutoscaler(version string) string {
+func testAccCheckScalewayK8SClusterConfigAutoscaler(version string) string {
 	return fmt.Sprintf(`
-resource "scaleway_k8s_cluster_beta" "autoscaler" {
+resource "scaleway_k8s_cluster" "autoscaler" {
```
Comment (kindermoumoute) [inline scaleway/resource_k8s_cluster_beta_test.go:257]: should keep `_beta`

File: scaleway/resource_k8s_cluster_beta_test.go [hunk 8]
```diff
@@ -451,98 +267,21 @@ resource "scaleway_k8s_cluster_beta" "autoscaler" {
...
-		placement_group_id = scaleway_instance_placement_group.pool_placement_group.id
-	}
-	tags = [ "terraform-test", "scaleway_k8s_cluster_beta", "default-pool-placement-group" ]
-}`, version)
-}
-
-func testAccCheckScalewayK8SClusterBetaDefaultPoolRecreate(nodeType string) string {
-	return fmt.Sprintf(`
-resource "scaleway_k8s_cluster_beta" "recreate_pool" {
+resource "scaleway_k8s_cluster" "auto_upgrade" {
```
Comment (kindermoumoute) [inline scaleway/resource_k8s_cluster_beta_test.go:276]: should keep `_beta`

File: scaleway/resource_k8s_cluster_beta_test.go (hunk 3 above)
Comment (Sh4d1) [reply scaleway/resource_k8s_cluster_beta_test.go:168]: Hmm shoud we keep the tests for a deprecated resource ? :thinking:

File: scaleway/resource_k8s_cluster_beta.go (hunk 2 above)
Comment (Sh4d1) [reply scaleway/resource_k8s_cluster_beta.go:491]: :thinking: This needs to be done here or else the Read will have no ID to read from :sweat_smile:

File: scaleway/resource_k8s_cluster_beta_test.go [hunk 9]
```diff
@@ -6,27 +6,61 @@ import (
...
+	k8s "github.com/scaleway/scaleway-sdk-go/api/k8s/v1"
 )
 
-func TestAccScalewayK8SClusterBetaMinimal(t *testing.T) {
+func TestAccScalewayK8SClusterDeprecated(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckScalewayK8SClusterBetaDestroy,
+		CheckDestroy: testAccCheckScalewayK8SClusterDestroy,
```
Comment (jerome-quere) [inline scaleway/resource_k8s_cluster_beta_test.go:16]: Do we want to keep beta here ?

File: website/docs/r/k8s_cluster_beta.html.markdown [hunk 10]
```diff
@@ -16,12 +16,15 @@ Creates and manages Scaleway Kubernetes clusters. For more information, see [the
...
-    size = 3
-  }
+  version = "1.18.0"
+  cni = "cilium"
+}
+
+resource "scaleway_k8s_pool_beta" "john" {
+  cluster_id = scaleway_k8s_cluster_beta.jack.id
+  name = "john"
+  node_type = "GP1-XS"
```
Comment (jerome-quere) [inline website/docs/r/k8s_cluster_beta.html.markdown:26]: Should we put DEV in example ?

File: scaleway/resource_k8s_pool_beta_test.go [hunk 11]
```diff
@@ -16,32 +16,42 @@ func TestAccScalewayK8SClusterPoolMinimal(t *testing.T) {
...
 				),
 			},
 			{
-				Config: testAccCheckScalewayK8SPoolBetaConfigMinimal("1.16.0", true),
+				Config: testAccCheckScalewayK8SPoolConfigMinimal("1.18.0", true),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckScalewayK8SClusterBetaExists("scaleway_k8s_cluster_beta.minimal"),
-					testAccCheckScalewayK8SPoolBetaExists("scaleway_k8s_pool_beta.minimal"),
+					testAccCheckScalewayK8SClusterExists("scaleway_k8s_cluster_beta.minimal"),
+					testAccCheckScalewayK8SPoolExists("scaleway_k8s_pool_beta.minimal"),
```
Comment (jerome-quere) [inline scaleway/resource_k8s_pool_beta_test.go:38]: The same remark about the beta for these two

**Pull request merged**